compiler/souschef/include/souschef/Data/Constant.h

```diff
@@ -62,6 +64,90 @@ template <typename T> struct ConstantDataChefFactory : public DataChefFactory
   }
 };
 
+class ConstantInt4DataChef final : public DataChef
+{
+public:
+  ConstantInt4DataChef(const int8_t &value) : _value{value}
+  {
+    // DO NOTHING
+  }
+
+public:
+  std::vector<uint8_t> generate(int32_t count) const override
+  {
+    std::vector<uint8_t> res;
+
+    if (_value < -8 || 7 < _value)
+      throw std::runtime_error("Constant value out of range.");
+
+    for (uint32_t n = 0; n < count; ++n)
+    {
+      const uint8_t *arr = reinterpret_cast<const uint8_t *>(&_value);
+
+      for (uint32_t b = 0; b < sizeof(int8_t); ++b)
+      {
+        res.emplace_back(arr[b]);
+      }
```

Comment:
For simplicity?

```suggestion
      const uint8_t val = reinterpret_cast<uint8_t>(_value);

      res.emplace_back(val);
```

Reply:
It would be possible to use `static_cast`.
```suggestion
      const uint8_t val = static_cast<uint8_t>(_value);
      res.emplace_back(val);
```

Reply:
I'll check

Reply:
I think this will do, as we're adding `uint8_t _value` into `std::vector<uint8_t> res`
```cxx
      res.emplace_back(_value);
```

Reply:
It's OK